Update balancing

bagustris opened the pull request:
- add more balancing methods (balancing now coded in balance.py instead of modelrunner):
            'ros',               # RandomOverSampler
            'smote',          # SMOTE
            'adasyn',         # ADASYN
            'borderlinesmote',                # BorderlineSMOTE
            'svmsmote'                             # SVMSMOTE
             'clustercentroids',                 # ClusterCentroids
            'randomundersampler',       # RandomUnderSampler
            'editednearestneighbours',  # EditedNearestNeighbours
            'tomeklinks'          # TomekLinks
             'smoteenn',          # SMOTEENN
            'smotetomek'       # SMOTETomek
- fix prediction path in mlp
- add more test files (by copilot):
   - mlp
   - xgb
   - whisper_transcriber
   - feats_praat_core
   - feats_opensmile
   -

felixbur commented:
It's new functionality, right?
Could you add example ini file to our examples that runs fast so we can check this still works with future changes?

or is this already there?

bagustris commented:
Added all examples in the examples directory. Also added a run script to run all these examples (using the Polish dataset).

```
chmod +x run_scripts/run_balancing_experiments.sh
bash run_scripts/run_balancing_experiments.sh
```

felixbur commented:
thanks @bagustris , i hope i got it: https://blog.syntheticspeech.de/2023/11/16/nkululeko-oversample-the-training-set/
